cdap-app-fabric/src/test/java/co/cask/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -234,9 +268,14 @@ public void testRunScheduledJobs() throws Exception {
     int runs2 = getRuns(WORKFLOW_2);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_1);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_2);
+
     publishNotification(dataEventTopic, WORKFLOW_1, AppWithFrequentScheduledWorkflows.DATASET_NAME1);
+    MessageId messageId = waitUntilProcessed(dataEventTopic, null);
```

Comment:
oh, right, this is the only test calling `publishNotification` and here is the first call. However, I feel waiting for number of jobs at the end will be faster than waiting for the notification one by one.